### ISSUE TITLE
Newest version has something wrong with GitHubRepository.php

### ISSUE THREAD

**Issue by cjmaio** [wppusher/wppusher-documentation#5]
When I installed, the version was telling me every theme and plugin couldn't be downloaded. Upon inspection of the `GitHubRepository.php` file I noticed that you have something hardcoded :wink: 

```
$url = 'https://api.github.com/repos/' . $this->handle . '/zipball/' . $this->getBranch() . '?dir=/wppusher';
```

I assume that was supposed to be configureable for the subdirectory option in each theme / plugin. Commenting that out (didn't want to try and fix it at the time) made it work.

This also happens with Bitbucket.

**Comment by petersuhm** [comment]
Thanks for reporting this @cjmaio. I'm gonna take a look to see what can have happened. It might be a permissions issue on your web server, but I'm not completely sure and need to debug it further. Do you have am existing `wp-content/upgrade` folder on your server?

Just to quickly explain: It's not really hard coded. It's a kind of "hack", to trick WordPress into downloading the repository in a directory called `upgrade/wppusher/something`, instead of `upgrade/random_repository_hash/something`. I'm working with a large WordPress hosting provider, to integrate WP Pusher better with their environment, and this is one of the steps towards that, so they always know where WP Pusher code goes.

**Comment by petersuhm** [comment]
Oh, one more thing. Do you install from a subdirectory in the repository? (shouldn't be related to this, but now I'm curious)

**Comment by cjmaio** [comment]
That was a quick response!

It looks like I do have an existing `wp-content/upgrade` folder on my server. It _should_ have permissions to overwrite it, as it's owned by www-data (it's an Ubuntu server) and has 744 permissions.

Okay, scratch that. It seems since it didn't have 775 permissions, it seems WordPress was erroring out. It appears it requires having the group write permission. Odd - another plugin must have created the upgrade folder, or perhaps it was WordPress...

**Comment by petersuhm** [comment]
If it's not too much trouble, I'd like to know it works if you delete the `/upgrade` folder. It's supposed to be wiped after each update anyways.

Thanks for helping debug this, @cjmaio!

**Comment by cjmaio** [comment]
I went and reverted my code changes that made it work... then tried to update an existing theme. It looks like it properly created the upgrade folder, and ran the upgrade. It did not, however, wipe the upgrade folder afterwards. It wiped the _contents_ of the folder, but the folder itself stayed.

I wonder if WordPress, or another plugin, created the upgrade folder with the odd 744 permissions.

Possibly the download error could check for incorrect permissions on the upload folder, and notify the user? When it initially occurred, I tried turning on logging and seeing if anything popped up there, but only saw the download failed error.
